pkg/receive/handler.go

```diff
@@ -91,3 +103,49 @@ func (h *Handler) Receive(w http.ResponseWriter, r *http.Request) {
 	h.forwardRequestsTotal.WithLabelValues("success").Inc()
 	w.WriteHeader(resp.StatusCode)
 }
+
+var ErrRequiredLabelMissing = fmt.Errorf("a required label is missing from the metric")
+
+// ValidateLabels makes sure that the request's content contains the required partitionKey
+func (h *Handler) ValidateLabels(next http.Handler) http.HandlerFunc {
```

Comment:
It doesn't seem like this really needs to be a method of the handler struct. IMO it should be an function that accepts a label name and an next and returns an HandlerFunc. This would also eliminate the need for the constructor func NewHandler to accept an extra argument.

Reply:
I guess the reason I did it that way was to cut down on the amount of args being passed around on the caller side.
Happy to change it though :blush:

Reply:
yes, I think it's better; these two things should be entirely separate concerns that are not coupled. It will also allow us to generalize the label requirement more: rather than enforcing `_id=$token` we can enforce any arbitrary `$key=$label`